src/d/actor/d_a_ship.cpp

```diff
@@ -5,460 +5,4718 @@
+        dStage_changeScene(iVar1, 0.0f, 0, 0xFF);
+    }
+    
+    return TRUE;
 }
 
 /* 00006FDC-0000713C       .text procStartModeWarp_init__8daShip_cFv */
-void daShip_c::procStartModeWarp_init() {
-    /* Nonmatching */
+BOOL daShip_c::procStartModeWarp_init() {
+    this->current.pos.y = this->m03F4 + 5000.0f;
+    onStateFlg(daSFLG_UNK1_e);
+    this->mProc = &daShip_c::procStartModeWarp;
+    this->m034C = 13;
+    if (this->m0392 != 10) {
+        setPartOffAnime();
+    }
+    this->m03A6 = 0x1C25;
+    camera_class* camera = dComIfGp_getCamera(dComIfGp_getPlayerCameraID(0));
+    camera->mCamera.Stop();
+    cXyz local_38(this->current.pos.x, this->m03F4 + 1500.0f, this->current.pos.z + 2000.0f);
+    camera->mCamera.Set(this->current.pos, local_38);
+    this->m037A = 0;
+    fopAcM_seStartCurrent(this, 0x186D, 0);
+    return TRUE;
 }
 
 /* 0000713C-000074D4       .text procStartModeWarp__8daShip_cFv */
-void daShip_c::procStartModeWarp() {
-    /* Nonmatching */
+BOOL daShip_c::procStartModeWarp() {
+    int iVar5;
+    _GXColor diff;
+    _GXColor amb;
+    cXyz local_30;
+    
+    daTornado_c* pfVar2 = (daTornado_c*)fopAcM_SearchByID(this->mTactWarpID);
+    camera_class* camera = dComIfGp_getCamera(dComIfGp_getPlayerCameraID(0));
+    this->shape_angle.y += this->m03A6;
+    this->current.angle.y = this->shape_angle.y;
+    if (pfVar2) {
+        if (checkStateFlg(daSFLG_UNK8_e) == 0) {
+            onStateFlg(daSFLG_UNK8_e);
+            pfVar2->current.pos.x = this->current.pos.x;
+            pfVar2->current.pos.z = this->current.pos.z;
+        }
+        iVar5 = 0;
+        for (int i = 0; i < 11; iVar5++, i++) {
+            if (this->current.pos.y < pfVar2->getJointYPos(i)) break;
+        }
+        if (iVar5 == 11) {
+          iVar5 = 10;
+        }
+        cLib_chaseF(&this->current.pos.x, pfVar2->getJointXPos(iVar5), 50.0f);
+        cLib_chaseF(&this->current.pos.z, pfVar2->getJointZPos(iVar5), 50.0f);
+        local_30.x = pfVar2->current.pos.x;
+        local_30.z = pfVar2->current.pos.z;
+    }
+    else {
+        local_30.x = this->current.pos.x;
+        local_30.z = this->current.pos.z;
+    }
+    local_30.y = this->current.pos.y;
+    camera->mCamera.Set(local_30, camera->mCamera.Eye());
+    if (this->m037A == 0) {
+        if (this->current.pos.y < this->m03F4) {
+            this->current.pos.y = this->m03F4 - 50.0f;
+            this->speed.y = 0.0f;
+            offStateFlg(daSFLG_UNK1_e);
+            this->m037A = 1;
+            if (pfVar2 != NULL) {
+                pfVar2->m31c = 1;
+            }
+            dKy_get_seacolor(&diff, &amb);
+            this->m03BC.x = 0;
+            this->m03BC.y = this->shape_angle.y + 0x8000;
+            this->m03BC.z = 0;
+            dComIfGp_particle_set(0x285, &this->current.pos, &this->shape_angle, NULL, 0xFF, &this->m1984, -1, &diff);
+            dComIfGp_particle_set(0x285, &this->current.pos, &this->m03BC, NULL, 0xFF, &this->m1998, -1, &diff);
```

Comment:
```suggestion
            dComIfGp_particle_set(dPa_name::ID_COMMON_0285, &this->current.pos, &this->shape_angle, NULL, 0xFF, &this->m1984, -1, &diff);
            dComIfGp_particle_set(dPa_name::ID_COMMON_0285, &this->current.pos, &this->m03BC, NULL, 0xFF, &this->m1998, -1, &diff);
```

Particle IDs have an enum in d_particle_name.h